lsp-mode.el

```diff
@@ -3584,20 +3584,22 @@ https://microsoft.github.io/language-server-protocol/specification#textDocument_
   (unless (seq-empty-p locations)
     (cl-labels ((get-xrefs-in-file
                  (file-locs location-link)
-                 (let* ((filename (seq-first file-locs))
-                        (visiting (lsp--buffer-for-file filename))
-                        (fn (lambda (loc)
-                              (lsp--xref-make-item filename
-                                                   (if location-link (or (gethash "targetSelectionRange" loc)
-                                                                         (gethash "targetRange" loc))
-                                                     (gethash "range" loc))))))
-                   (if visiting
-                       (with-current-buffer visiting
-                         (seq-map fn (cdr file-locs)))
-                     (when (file-readable-p filename)
-                       (with-temp-buffer
-                         (insert-file-contents-literally filename)
-                         (seq-map fn (cdr file-locs))))))))
+                 (let ((filename (seq-first file-locs)))
+                   (if (equal filename "")
```

Comment:
Instead of checking whether the file is "" which looks more like a patch can you change the code to a more generic solution, something like:

``` emacs-lisp
(conditional-case err 
  ;;;   do processing 
 (error (lsp--warn "Failed to process xref entry with message: %s" (error-message-string err))
 (file-error (lsp--warn ...
```

the point is here to guard against any invalid files not only "".

Reply:
I updated the function to be more generic.